main_test.go

```diff
@@ -461,13 +461,17 @@ func TestMatchFuzzyError(t *testing.T) {
 	t.Run("ambiguous migration: two different items have the same match", func(t *testing.T) {
 		create := set.NewStringSet(`abcde`, `abdecde`)
 		destroy := set.NewStringSet(`abdcde`, `hfjabd`)
-		wantError := "ambiguous migration: {abcde} -> {abdcde} or {abdecde} -> {abdcde}"
 		_, _, err := matchFuzzy(create, destroy)
 		if err == nil {
 			t.Fatalf("got: no error; want: an ambiguous migration error")
 		}
-		if err.Error() != wantError {
-			t.Fatalf("got: %s; want: %s", err.Error(), wantError)
+		switch {
+		case !strings.HasPrefix(err.Error(), "ambiguous migration:"):
+			t.Fatalf("error: %s; should start with 'ambiguous migration:' prefix", err.Error())
+		case !strings.Contains(err.Error(), "{abcde} -> {abdcde}"):
+			t.Fatalf("error: %s; should contain '{abcde} -> {abdcde}'", err.Error())
+		case !strings.Contains(err.Error(), "{abdecde} -> {abdcde}"):
+			t.Fatalf("error: %s; should contain '{abdecde} -> {abdcde}'", err.Error())
```

Comment:
The `switch` is confusing and not needed in test code. Also, we don't want t.Fatal here, we want t.Error, because we want to report all the possible errors. Using t.Fatal stops the test.

We also want to report the error message only once, not 3 times.

So we need to build our error report as a string, something like:

We can simply write something like

```go

gotMsg := err.Error()
var string msg

want := "ambiguous migration:"
if !strings.HasPrefix(gotMsg, want) {
    msg += fmt.Sprintf("error message does not start with %q\n", want)
}

want = "{abcde} -> {abdcde}"
if !strings.Contains(gotMsg, want):
    msg += fmt.Sprintf("error message does not contain %q", want)
}

// the 3rd check here


if msg != "" {
    t.Fatal(msg)
}
```

Reply:
Changed as suggested.